src/molecular-weight.jl

```diff
@@ -0,0 +1,260 @@
+const AA_X_MW_MW  = 110                       # Any amino acid (average)    ->  Source = http://thermofisher.com/de/de/home/references/ambion-tech-support/rna-tools-and-calculators/proteins-and-amino-acids.html
+const AA_O_Gap_MW = 0                         # Consider gap as 0       
+
+# Defining the function molecular_weight, which accepts a AA sequence and units ("gmol", "KDa") as input and return molecular weight in either g/mol or kDa. If not specified it returns in g/mol
+function molecular_weight(aa_seq::AASeq, unit=:"gmol")
+    AA_vector = Vector{Float64}(undef, 0)
+    i = 1
+    while i <= length(aa_seq)
+        if aa_seq[i] == AA_A
+            push!(AA_vector, AA_A_MW)
+        elseif aa_seq[i] == AA_R
+            push!(AA_vector, AA_R_MW)
+        elseif aa_seq[i] == AA_N
+            push!(AA_vector, AA_N_MW)
+        elseif aa_seq[i] == AA_D
+            push!(AA_vector, AA_D_MW)
+        elseif aa_seq[i] == AA_C
+            push!(AA_vector, AA_C_MW)
+        elseif aa_seq[i] == AA_Q
+            push!(AA_vector, AA_Q_MW)
+        elseif aa_seq[i] == AA_E
+            push!(AA_vector, AA_E_MW)
+        elseif aa_seq[i] == AA_G
+            push!(AA_vector, AA_G_MW)
+        elseif aa_seq[i] == AA_H
+            push!(AA_vector, AA_H_MW)  
+        elseif aa_seq[i] == AA_I
+            push!(AA_vector, AA_I_MW)
+        elseif aa_seq[i] == AA_L
+            push!(AA_vector, AA_L_MW)    
+        elseif aa_seq[i] == AA_K
+            push!(AA_vector, AA_K_MW)
+        elseif aa_seq[i] == AA_M
+            push!(AA_vector, AA_M_MW)                                            
+        elseif aa_seq[i] == AA_F
+            push!(AA_vector, AA_F_MW)
+        elseif aa_seq[i] == AA_P
+            push!(AA_vector, AA_P_MW)
+        elseif aa_seq[i] == AA_S
+            push!(AA_vector, AA_S_MW)
+        elseif aa_seq[i] == AA_T
+            push!(AA_vector, AA_T_MW)   
+        elseif aa_seq[i] == AA_W
+            push!(AA_vector, AA_W_MW)
+        elseif aa_seq[i] == AA_Y
+            push!(AA_vector, AA_Y_MW)
+        elseif aa_seq[i] == AA_V
+            push!(AA_vector, AA_V_MW)
+        elseif aa_seq[i] == AA_O
+            push!(AA_vector, AA_O_MW_MW)
+        elseif aa_seq[i] == AA_U
+            push!(AA_vector, AA_U_MW_MW)
+        elseif aa_seq[i] == AA_B
+            push!(AA_vector, AA_B_MW_MW)
+        elseif aa_seq[i] == AA_J
+            push!(AA_vector, AA_J_MW_MW)
+        elseif aa_seq[i] == AA_Z
+            push!(AA_vector, AA_Z_MW_MW)
+        elseif aa_seq[i] == AA_X
+            push!(AA_vector, AA_X_MW_MW)
+        elseif aa_seq[i] == AA_O_Gap
+            push!(AA_vector, AA_O_Gap_MW)                           
+        else
+            println("Error due to unknown amino acid at $i position")
+            return nothing
+        end
+        i = i+1
+    end
+        if unit == "gmol"
+            return round(sum(AA_vector) - ((length(aa_seq) - 1) * WATER_MW); digits = 3)
```

Comment:
I would not round the result, give it to the user with all decimals. Even if the input weights themselves are not known that precisely, rounding is only really useful when displaying it to the user in the final step, and should not be done here, when computing and returning the value.
In other words, rounding is for "human consumption" and should be done when displaying. Programs are perfectly happy with all the decimals you can fit in a 64-bit float.